pydantic/types.py

```diff
@@ -527,46 +528,36 @@ def condecimal(
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ UUID TYPES ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 
-@_dataclasses.dataclass(**_internal_dataclass.slots_true)
-class UuidVersion:
-    uuid_version: Literal[1, 3, 4, 5]
+@_dataclasses.dataclass
+class Version(_fields.PydanticMetadata, BaseMetadata):
+    """A field metadata class to indicate that a field should be validated in version mode.
 
-    def __get_pydantic_json_schema__(
-        self, core_schema: core_schema.CoreSchema, handler: _annotated_handlers.GetJsonSchemaHandler
-    ) -> JsonSchemaValue:
-        field_schema = handler(core_schema)
-        field_schema.pop('anyOf', None)  # remove the bytes/str union
-        field_schema.update(type='string', format=f'uuid{self.uuid_version}')
-        return field_schema
+    Attributes:
+        version: Whether to validate the field in version mode.
 
-    def __get_pydantic_core_schema__(
-        self, source: Any, handler: _annotated_handlers.GetCoreSchemaHandler
-    ) -> core_schema.CoreSchema:
-        return core_schema.general_after_validator_function(
-            cast(core_schema.GeneralValidatorFunction, self.validate), handler(source)
-        )
+    Example:
+        ```python
+        from typing_extensions import Annotated
 
-    def validate(self, value: UUID, _: core_schema.ValidationInfo) -> UUID:
-        if value.version != self.uuid_version:
-            raise PydanticCustomError(
-                'uuid_version', 'uuid version {required_version} expected', {'required_version': self.uuid_version}
-            )
-        return value
+        from pydantic.types import Version
+
+        UuidVersion = Annotated[int, Version]
+        ```
+    """
+
+    version: int
 
     def __hash__(self) -> int:
-        return hash(type(self.uuid_version))
+        return hash(self.version)
```

Comment:
cc @adriangb I would be grateful to hear your input here on what you is appropriate to do with this type. My instinct is against adding a generic `Version` type and keeping it just as `UuidVersion`, maybe there's a technical reason I've missed for doing this. We should be careful not to create a breaking change.